docs/source/llms/langchain/autologging.rst

```diff
@@ -0,0 +1,262 @@
+
+.. attention::
+
+    MLflow's LangChain Autologging feature has been overhauled in the ``MLflow 2.14.0`` release. If you are using the earlier version of MLflow, please refer to the legacy documentation `here <#documentation-for-old-versions>`_ for applicable autologging documentation.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 1
+
+Quickstart
+----------
+
+To enable autologging for LangChain models, call :py:func:`mlflow.langchain.autolog()` at the beginning of your script or notebook. This will automatically log the traces by default as well as other artifacts such as models, datasets, and model signatures if you explicitly enable them. For more information about the configuration, please refer to the `Configure Autologging <#configure-autologging>`_ section.
+
+.. code-block::
+
+    import mlflow
+
+    mlflow.langchain.autolog()
+
+    # Enable other optional logging
+    # mlflow.langchain.autolog(log_models=True, log_datasets=True)
+
+    # Your LangChain model code here
+    ...
+
+Once you have invoked the chain, you can view the logged traces and artifacts in the MLflow UI.
+
+.. figure:: ../../_static/images/llms/tracing/langchain-tracing.gif
+    :alt: LangChain Tracing via autolog
+    :width: 100%
+    :align: center
+
+
+Configure Autologging
+---------------------
+
+MLflow LangChain autologging can log various information about the model and its inference. **By default, only trace logging is enabled**, but you can enable autologging of other information by setting the corresponding parameters when calling :py:func:`mlflow.langchain.autolog()`. For other configurations, please refer to the API documentation.
+
+.. list-table::
+    :widths: 20 20 30 30
+    :header-rows: 1
+
+    * - Target
+      - Default
+      - Parameter
+      - Description
+    * - Traces
+      - ``true``
+      - ``log_traces``
+      - Whether to generate and log traces for the model. See `MLflow Tracing <../tracing/index.html>`_ for more details about tracing feature.
+    * - Model Artifacts
+      - ``false``
+      - ``log_models``
+      - If set to ``True``, the LangChain model will be logged when it is invoked. Supported models are `Chain`, `AgentExecutor`, `BaseRetriever`, `RunnableSequence`, `RunnableParallel`, `RunnableBranch`, `SimpleChatModel`, and `ChatPromptTemplate` (Additional model types will be supported in the future.)
+    * - Model Signatures
+      - ``false``
+      - ``log_model_signatures``
+      - If set to ``True``, :py:class:`ModelSignatures <mlflow.models.ModelSignature>` describing model inputs and outputs are collected and logged along with Langchain model artifacts during inference.
+    * - Input Example
+      - ``false``
+      - ``log_input_examples``
+      - If set to ``True``, input examples from inference data are collected and logged along with LangChain model artifacts during inference.
```

Comment:
Note: Input examples are MLflow model attributes and are only collected if ``log_models`` is also ``True``.